### PR TITLE
feat(core): honor declared severity from lesson prose in compile output (#1656)

### LINKED ISSUE #1656: core: compile LLM emits 'severity: warning' even when lesson prose declares 'Severity: error'
Filed from strategy-side upstream-feedback item at `.strategy/upstream-feedback/015-compile-default-severity-ignores-lesson-prose.md` (full write-up lives there; this issue is the thin tracking mirror).

## Signal

Lessons with explicit `Severity: error` prose compile out to rules with `"severity": "warning"` in `compiled-rules.json`. The LLM ingests the prose but does not honor it.

## Evidence

From `mmnto-ai/liquid-city#77` session-17, compile log stamps:

```
[ADR-008 Rule 2: RunManager must not mutate TacticalState] Compiled (regex, warning): /\b(?:ResMut\s*<\s*TacticalState\s*>|&\s*mut\s*TacticalState\b)/
[ADR-008 Rule 3a: run_manager.rs must not call TacticalDirector-owned functions] Compiled (regex, warning): /\b(tactical_director_tick|update_heat|apply_host_authoritative_heat|advance_sim_tick_count)\s*\(/
```

The `(regex, warning)` stamp is the LLM's emitted severity, not a totem default applied post-hoc. Five rules in this session, all with the same mismatch against their source lessons.

Sample lesson prose the LLM ingested (from `.totem/lessons/lesson-ef0ba0d7.md`):

> **Severity: error.** Violations indicate a direct breach of the publisher/consumer contract the ADR-008 decomposition is built on.

## Downstream cost

5 manual `"severity"` edits per compile round + 1 `totem lesson compile --refresh-manifest` invocation. On `mmnto-ai/liquid-city#77` this happened twice (R2 and R3), burning ~10 edits total that served no authoring purpose. The contract between lesson prose and compiled rule is broken; every compile cycle re-applies the mechanical fix.

Further context: per a CR review learning cited in the upstream-feedback item, `mmnto-ai/totem` collects only `severity: "error"` as CI-blocking — `warning` is advisory. So a warning-severity rule silently degrades to advisory-only governance even when the author declared it an architectural-invariant error.

## Proposed approach

**Option A (recommended for 1.16.0).** Parse `Severity: <level>` from lesson prose at compile time (case-insensitive, tolerant of bold/italic markdown like `**Severity:** error`). Pass the result as a structured parameter to the compile prompt alongside `tags` / `scope` / `fileGlobs`, which are already structured. When the lesson declares `Severity: error`, the compile output emits `"severity": "error"`. Silent lesson falls back to current `warning` default.

**Option B (1.17.0).** First-class `severity:` frontmatter field on the lesson schema. Schema migration + MCP `add_lesson` signature update. Cleaner long-term but breaks the "plain markdown" authoring ergonomic.

**Option C (bridge).** Emit a compile-time warning when a lesson contains a `Severity: error` prose line but the compiled rule emits `warning`. Surfaces the mismatch earlier but doesn't fix it.

## Class interaction

- Adjacent to #1653 (α engine dispatch) and #1654 (β compile-pipeline grammar). Both items surface from the same compile step; both are "compile pipeline doesn't honor what the contributor wrote." 1653/1654 are engine/grammar surface; this is severity-level metadata surface.
- Adjacent to #1643 (structured source-provenance to lesson frontmatter). If #1643 ships first, Option B folds into its schema work.

## Severity

Tier-2. Recurring friction on every compile cycle in any repo where lessons use `Severity: error` prose convention. Not blocking — contributors can archive-in-place. But the archive-in-place cost scales with lesson volume, and the mechanical 5-10 edits per round serve no authoring purpose.

## Acceptance

- [ ] `parseSeverityFromProse(body: string): 'error' | 'warning' | null` helper in core.
- [ ] Compile prompt receives severity as structured input when prose declares it.
- [ ] Regression test: lesson with `**Severity:** error` in prose compiles to `"severity": "error"`.
- [ ] Regression test: lesson with `Severity: error.` (trailing period) compiles to `"severity": "error"`.
- [ ] Regression test: lesson with no severity marker falls back to `"warning"` default.
- [ ] Regression test: the five liquid-city ADR-008 exhibits compile with `"severity": "error"` when their prose is unchanged.

Upstream-feedback write-up: `.strategy/upstream-feedback/015-compile-default-severity-ignores-lesson-prose.md`

### PR DESCRIPTION
**Pull request by satur8d**
## Summary

Close the gap between the already-parsed severity in `lesson.frontmatter` and the compiled rule's emitted severity. The prose-severity parser has been in the codebase since the legacy frontmatter shim (`buildFrontmatterFromLegacy`); the compile pipeline was reading `parsed.severity` from the LLM output and discarding the authoritative declaration. This PR plumbs the declared value through at two layers.

## LC impact

Five ADR-008 rules on `mmnto-ai/liquid-city#77` burned ~10 manual severity-edit commits across R2 + R3 rounds because the compile pipeline emitted `"severity": "warning"` despite the lesson prose declaring `Severity: error`. After this PR merges, the next `totem lesson compile` cycle on liquid-city emits declared severity directly — the archive-in-place severity-edit loop goes away.

## Fix surfaces (two layers)

- **Compile prompt directive.** Both `COMPILER_SYSTEM_PROMPT` and `PIPELINE3_COMPILER_PROMPT` gain a `### Declared Severity (mmnto-ai/totem#1656)` section instructing the LLM to honor `**Severity:** error` / `Severity: warning` prose declarations in the emitted JSON. Reduces the frequency of drift at source.
- **Post-LLM override in `buildCompiledRule`.** Deterministic safety net that replaces `parsed.severity` with the declared value when they differ. LLM compliance is belt-and-suspenders, not load-bearing.

## Telemetry for prompt-tuning feedback

When the override actually changes the final severity, a record tagged `type: 'severity-override'` appends to `.totem/temp/telemetry.jsonl` via the new `onSeverityOverride` callback on `CompileLessonCallbacks`. CLI `compile.ts` wires the sink closure over the cwd-aware `totemDir` (matches the `mmnto-ai/totem#1645` pattern). 

**Marker-fire logic** (Shield finding from the review round): fires only when `declaredSeverity` differs from `emittedSeverity ?? 'warning'`, NOT from `emittedSeverity` alone. Prevents telemetry noise when a lesson declares `'warning'` and the LLM omits severity entirely (both resolve to the default; nothing changed).

## What changed

- **`packages/core/src/lesson-pattern.ts`** — new `parseDeclaredSeverity(body)` helper. Reuses `extractField` for prose extraction. Returns normalized `'error' | 'warning' | undefined`.
- **`packages/core/src/compile-lesson.ts`** — `BuildCompiledRuleOptions.declaredSeverityOverride`, `BuildRuleResult.severityOverride`, `CompileLessonCallbacks.onSeverityOverride`. `compileLesson` extracts once near the top and threads into both internal `buildCompiledRule` calls (Pipeline 2 and Pipeline 3).
- **`packages/cli/src/commands/compile.ts`** — `writeSeverityOverrideTelemetry` closure near `compileCommand` init. Wired as `onSeverityOverride` for the local path; called inline on the cloud path after `buildCompiledRule` returns.
- **`packages/cli/src/commands/compile-templates.ts`** — Declared Severity section appended to both prompts after the Test-Contract Scope Classifier.
- **`packages/core/src/index.ts`** — `parseDeclaredSeverity` re-export.
- **`.totem/specs/1656.md`** — preflight spec + Implementation Design. Captures the architectural insight that the parser already existed; the ticket was plumbing, not authoring.
- **`.strategy` submodule bump** `9e38d44` → `113179c`. Picks up strategy PR #122 (upstream-feedback 012/013 filed frontmatter) and strategy PR #125 (items 015 + 016 from liquid-city session-17).

## Tests (11 new, all passing)

- **5 in `compile-lesson.test.ts`** — `buildCompiledRule` with / without `declaredSeverityOverride` across the matrix (LLM emits different / same / none / warning-matches-default).
- **9 in `lesson-pattern.test.ts`** — `parseDeclaredSeverity` edge cases (canonical, case, alt markdown, bare, multi-declaration, out-of-vocab, trailing-punctuation boundary, free-prose anti-trigger).
- **7 in `compile-templates.test.ts`** — "Declared Severity" directive present on both prompts, issue reference, testing-tag signal, severity values emitted.

Full suites: core **1,331 / 1,331**, cli **1,819 / 1,819**.

## Gates

- [x] `pnpm run format`
- [x] `pnpm exec totem lint` PASS (0 errors, 8 warnings — all #1626-class false positives pending retriage)
- [x] `pnpm exec totem review` PASS (0 findings after DRY refactor + logic fix)
- [x] `pnpm exec totem verify-manifest` PASS (443 rules)
- [x] Pre-push hook PASS
- [ ] Bot review (CR + GCA)
- [ ] CI green
- [ ] Follow-up chore PR retriages LC's five ADR-008 rules via `totem compile --upgrade <hash>`

Closes #1656

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/core/src/lesson-pattern.ts [hunk 1]
```diff
@@ -47,6 +47,22 @@ export interface ManualPattern {
...
+ * Reuses `extractField` for the prose-extraction rules (markdown-bold
+ * tolerance, mandatory colon, case-insensitive match, line-anchored). Added
+ * as a shared helper so the compile pipeline's declared-severity override
+ * (in `compile-lesson.ts::compileLesson`) and the CLI's cloud-path override
+ * (in `compile.ts`) normalize from a single source of truth.
+ */
+export function parseDeclaredSeverity(body: string): 'error' | 'warning' | undefined {
+  const raw = extractField(body, 'Severity')?.toLowerCase();
+  return raw === 'error' || raw === 'warning' ? raw : undefined;
+}
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/core/src/lesson-pattern.ts:86]: Fixed in `772e7f2c`. `parseDeclaredSeverity` now strips leading and trailing markdown markers (`*`, `_`) and trailing sentence punctuation (`.`, `,`, `;`, `:`, `!`, `?`) in a single pass before the strict enum match. Covered shapes: `**Severity:** error.`, `**Severity: error**`, `Severity: **error**`, `Severity: **error**.`, and ```Severity: `error````. Regression tests added in `lesson-pattern.test.ts` locking in each shape plus the out-of-vocabulary rejection (`info`, `critical` still return undefined).

**Comment by satur8d** [comment]
@gemini-code-assist Thanks for the review. All three findings fixed in `772e7f2c`:

**1. `parseDeclaredSeverity` too strict (GCA high / CR major).** Confirmed — the previous implementation returned `undefined` for common prose shapes like `**Severity:** error.` and `**Severity: error**`. Updated the normalization to strip backtick wrappers, leading and trailing markdown markers (`*`, `_`), and trailing sentence punctuation (`.`, `,`, `;`, `:`, `!`, `?`) in a single pass:

```ts
const normalized = stripInlineCode(raw.trim())
  .replace(/^[*_]+|[*_.,;:!?]+$/g, '')
  .trim()
  .toLowerCase();
```

Shield round-2 caught a follow-on: leading markers weren't being stripped in my first pass. The combined regex above closes both ends in one operation. Regression tests added in `packages/core/src/lesson-pattern.test.ts` locking in each observed shape:

- `**Severity:** error.` (trailing period)
- `**Severity: error**` (full-line bold)
- `Severity: **error**` (value-only bold)
- `Severity: **error**.` (combined bold + punctuation)
- `` **Severity:** `error` `` (backtick-wrapped value)

Strict enum equality still follows the strip, so `**Severity:** info` and `**Severity:** critical` continue to return `undefined` — out-of-vocabulary declarations preserve the compile-pipeline's `'warning'` default fallback.

**2. JSON schema examples missing `severity` field (CR info, 239-252 and 631-644).** Fair point — the directive was added in prose but the schema examples omitted the field, which weakens the in-context signal. Added `"severity": "warning"` to both `COMPILER_SYSTEM_PROMPT` Output Schema blocks and to the `PIPELINE3_COMPILER_PROMPT` example. The default `"warning"` in the examples nudges the LLM toward emitting the field with the expected default when no declaration is present in the lesson prose.

**Comment by satur8d** [comment]
@coderabbitai full review

**Comment by satur8d** [comment]
@gemini-code-assist Round-3 finding fixed in `44f0920f`. You're right — `stripInlineCode` only matches when backticks are at the absolute string edges, so running it before the markdown/punctuation strip left backticks in place on bold-wrapped and punctuation-trailing shapes:

- `**` + backtick-wrapped `error` + `**` (bold-wrapped backticks)
- backtick-wrapped `error` + `.` (backticks followed by trailing period)
- `**` + backtick-wrapped `warning` + `**` + `.` (both)

Order swapped: markdown / punctuation strip first (on the bare trimmed value), then `stripInlineCode` on the isolated value. All three shapes now normalize correctly because the backticks land at the string edges after the outer strip runs.

Three regression tests added in `packages/core/src/lesson-pattern.test.ts` covering the three shapes named in your comment — all pass. The lowercase normalization was already at the end of the chain; the core fix is just the order swap.

File: packages/core/src/compile-lesson.ts [hunk 2]
```diff
@@ -370,7 +392,27 @@ export function buildCompiledRule(
...
+  // the final severity is 'warning' both ways and the marker stays absent
+  // (Shield finding on initial PR round).
+  const declaredSeverity = options.declaredSeverityOverride;
+  const emittedSeverity = parsed.severity;
+  const severity: 'error' | 'warning' = declaredSeverity ?? emittedSeverity ?? 'warning';
+  const wouldHaveShipped = emittedSeverity ?? 'warning';
+  const severityOverride =
+    declaredSeverity !== undefined && declaredSeverity !== wouldHaveShipped
+      ? { from: emittedSeverity, to: declaredSeverity }
+      : undefined;
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/core/src/compile-lesson.ts:414]: Fixed in `2f5d3eef`. You were right — the marker was only attached to the success return at the end of `buildCompiledRule`, so every early rejection path dropped the signal and telemetry under-counted drift exactly where it would be most informative.

Added a `reject(rejectReason)` helper per your suggestion that threads `severityOverride` through when present:

```ts
const reject = (rejectReason: string): BuildRuleResult =>
  severityOverride
    ? { rule: null, rejectReason, severityOverride }
    : { rule: null, rejectReason };
```

All early-exit paths in `buildCompiledRule` now call `reject(...)` instead of constructing the bare `{ rule: null, rejectReason }` object: the ast-grep validator reject, the AST-engine missing-field reject, the regex validator reject, the suppression-directive reject on both ast-grep and regex paths, and the four smoke-gate rejects (missing badExample, zero-match badExample, missing goodExample, matches goodExample).

`buildManualRule` is left untouched since it doesn't accept `declaredSeverityOverride` (Pipeline 1 manual rules bypass the prose-severity override).

Two regression tests added:

- Rejection path + LLM drift: marker is present on the rejected result.
- Rejection path + matching severity: marker stays absent (no false fires).

**Comment by satur8d** [comment]
@gemini-code-assist @coderabbitai ready for final review. All findings from prior rounds addressed:

| Round | Finding | Fix SHA |
|---|---|---|
| R1 | GCA high / CR major — `parseDeclaredSeverity` too strict for common prose shapes (trailing period, full-line bold, bold-wrapped value) | `772e7f2c` |
| R1 | CR info — Output Schema examples omit `"severity"` field, weakening the directive | `772e7f2c` + `466603fa` (extended to the five concrete Lesson → Output pairs in the Examples section) |
| R2 | GCA high — `stripInlineCode` called before markdown/punctuation strip leaves backticks in place on bold-wrapped or punctuation-trailing shapes | `44f0920f` |
| R3 | CR minor — `severityOverride` marker only attached to success return; rejection paths dropped the signal and undercounted drift | `2f5d3eef` |

Current branch state:

- 148 core tests passing (+11 new for parseDeclaredSeverity edge cases, +4 new for buildCompiledRule rejection-path behavior, +3 new for the bot-round-2 backtick + markdown combinations)
- `totem lint` PASS, 0 errors
- `totem review` PASS, 0 findings
- Format clean, manifest verified

Final summary of what landed on the branch:

1. `parseDeclaredSeverity(body)` helper in `@mmnto/totem` that normalizes prose-declared severity against common markdown and punctuation shapes.
2. `buildCompiledRule` honors a `declaredSeverityOverride` option and reports the override event in `BuildRuleResult.severityOverride`, on both success and rejection paths.
3. `CompileLessonCallbacks.onSeverityOverride` callback for CLI telemetry; CLI `compile.ts` wires a `writeSeverityOverrideTelemetry` closure over the cwd-aware `totemDir`.
4. Declared Severity directive sections on both `COMPILER_SYSTEM_PROMPT` and `PIPELINE3_COMPILER_PROMPT`, plus `"severity": "warning"` mirrored in every Output Schema example and the five concrete Lesson → Output few-shot examples.
5. `.strategy` submodule bump to pick up upstream-feedback items 012, 013, 015, 016 frontmatter-filed.

Closes #1656.
